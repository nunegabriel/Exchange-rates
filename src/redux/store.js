import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import CoinReducer from './coin/home';

const store = configureStore({
  reducer: {
    cryptoCoin: CoinReducer,
    middleware: [thunk],
  },
});

export default store;
