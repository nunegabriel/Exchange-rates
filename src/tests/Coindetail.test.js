
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import Detail from '../components/Coindetail';
import '@testing-library/jest-dom';
import store from '../redux/store';

it('Detail page testing', () => {
  const detail = renderer.create(
    <Provider store={store}>
      <Router>
        <Detail />
      </Router>
    </Provider>,
  )
  expect(detail).toMatchSnapshot();
});
