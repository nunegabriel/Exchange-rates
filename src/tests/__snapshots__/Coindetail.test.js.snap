// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail page testing 1`] = `
Array [
  <a
    href="/"
    onClick={[Function]}
  >
    <p
      className="exit-div"
    >
      <i
        className="Exit"
        label="next"
        role="button"
        tabIndex={0}
      />
    </p>
    <div>
      <h1
        className="logo"
      >
        CoinX
      </h1>
    </div>
  </a>,
  <table>
    <div
      className="crypto-div"
    />
  </table>,
]
`;
